cmd/controller/main.go

```diff
@@ -433,6 +444,55 @@ type httpServer struct {
 	http.ServeMux
 }
 
+func runLeaderElection(ctx context.Context, cancel context.CancelFunc, logger *logrus.Entry, ctlConf config, kubeClient *kubernetes.Clientset, start func(_ context.Context)) {
+	if !ctlConf.LeaderElection {
+		start(ctx)
+	} else {
+		id := uuid.New().String()
+
+		lock := &resourcelock.LeaseLock{
+			LeaseMeta: metav1.ObjectMeta{
+				Name:      "agones-controller-lock",
+				Namespace: "agones-system",
+			},
+			Client: kubeClient.CoordinationV1(),
+			LockConfig: resourcelock.ResourceLockConfig{
+				Identity: id,
+			},
+		}
+
+		logger.Infof("This controller ID is %s", id)
```

Comment:
nit: logrus style is weird and I'm not sure I like it, but in general this should be something like:

logger.WithField("id", id).Info("Leader election ID")

Reply:
Done! and for the other ones as well.